src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java

```diff
@@ -102,17 +105,21 @@ public String getIssueDescription(final IssueCategory issueCategory, final Strin
         return issueDescription.toString();
     }
 
-    public String createVulnerableComponentsLink(final String componentName, final String baseVulnerableComponentsLink) {
-        if (StringUtils.isNotBlank(componentName)) {
-            final String encodedComponentName;
+    public Optional<String> createUrlWithComponentNameQuery(final String url, final String componentName) {
+        if (StringUtils.isNotBlank(componentName) && StringUtils.isNotBlank(componentName)) {
             try {
-                encodedComponentName = URLEncoder.encode(componentName, "UTF-8");
-                return String.format("%s?q=componentName:%s", baseVulnerableComponentsLink, encodedComponentName);
-            } catch (UnsupportedEncodingException e) {
-                logger.debug("Error encoding vulnerable components link: " + e.getMessage());
+                // Black Duck does not encode query parameters in the traditional way (UTF-8), so we must do a little extra work to ensure the encoding will match.
+                final String unencodedUrl = String.format("%s?q=componentName:%s", url, componentName);
+                URL encodedUrl = new URL(unencodedUrl);
+                URI uri = new URI(encodedUrl.getProtocol(), encodedUrl.getUserInfo(), IDN.toASCII(encodedUrl.getHost()), encodedUrl.getPort(), encodedUrl.getPath(), encodedUrl.getQuery(), encodedUrl.getRef());
+
+                final String asciiString = uri.toASCIIString();
```

Comment:
Do you want to include the stack overflow link in a comment to explain this method of encoding?

Reply:
I've unfortunately had to do the same thing before to ensure proper encoding as well in the past.